common/app/common/ApplicationMode.scala

```diff
@@ -0,0 +1,19 @@
+package common
+
+import play.api.{Application, GlobalSettings}
+import play.api.Mode._
+
+object ApplicationMode {
+  private var theMode = Prod
```

Comment:
it could be a global object, that supports DI-injected Application. Slightly perverse, but it means we can actually make a choice about how we want to use DI, rather than have it forced upon us.